functions.php

```diff
@@ -7,20 +7,110 @@
 function databaseConnect(): PDO {
     $db = new PDO('mysql:host=db; dbname=collection_project', 'root', 'password');
     $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_WARNING);
     return $db;
 }
+
 /**
  * Creates query fetching all data from the database
  *
  * @param PDO created by connection with database
  * @return array returns all of the data requested from the database
  */
-function databaseFetchAll($db): array{
-    $query = $db->prepare("SELECT `charname`, `class`, `level`, `strength`, `dexterity`, `constitution`, `intelligence`, `wisdom`, `charisma` FROM `characters`;");
+function databaseFetchAll(PDO $db): array {
+    $query = $db->prepare("SELECT `charname`, `class`, `level`, `strength`, `dexterity`, `constitution`, `intelligence`, `wisdom`, `charisma`, `id` FROM `characters` WHERE `deleted` = 0;");
     $query->execute();
     return $query->fetchAll();
 }
 
+/**
+ * Fetch character data from database so that it can be edited
+ *
+ * @param PDO database PDO
+ * @param string id of the character in the database
+ * @return array Returns all the information of the character that is to be edited
+ */
+function databaseFetchEditChar(PDO $db,string $id): array {
+    $query = $db->prepare("SELECT `id`, `charname`, `class`, `level`, `strength`, `dexterity`, `constitution`, `intelligence`, `wisdom`, `charisma` FROM `characters` WHERE `id` = :id ;");
+    $query->bindParam(':id', $id);
+    $query->execute();
+    return $query->fetch();
+}
+
+/**
+ * Sets the character data on the database to be tagged as deleted
+ *
+ * @param PDO database PDO
+ * @param string id of the character to be deleted
+ * @return returns a boolean to see if the function has failed
+ */
+function databaseDeleteChar(PDO $db, string $id) {
+    $query = $db->prepare("UPDATE `characters` SET `deleted` = 1 WHERE `id` = :id");
+    $query->bindParam(':id', $id);
+    return $query->execute();
```

Comment:
Similar to my comment about your edit function, if you pass in an ID that doesn't exist, this function actually still returns true.

What you can do is check $query->rowCount() (which will let you know how many rows where edited), so you could check that to make your function a bit more reliable